components/places/ffi/src/lib.rs

```diff
@@ -460,7 +482,7 @@ pub extern "C" fn bookmarks_get_tree(
 ) -> ByteBuffer {
     log::debug!("bookmarks_get_tree");
     CONNECTIONS.call_with_result(error, handle, |conn| -> places::Result<_> {
-        let root_id = SyncGuid(guid.into());
+        let root_id = SyncGuid::from(guid.into_string());
```

Comment:
I think `SyncGuid::from(guid.as_str());` works. Please use that for all creation of `SyncGuid` from `FfiStr`.